test/suites/auth.sh

```diff
@@ -79,11 +79,33 @@ test_authorization() {
   ! lxc auth identity group add oidc/test-user@example.com not-found || false # Group not found
   lxc auth identity group add oidc/test-user@example.com test-group # Valid
 
-  # Check user has been added to the group.
+  # Test fine-grained TLS identity creation
+  tls_identity_token="$(lxc auth identity create tls/test-user --quiet --group test-group)"
+  LXD_CONF2=$(mktemp -d -p "${TEST_DIR}" XXX)
+  LXD_CONF="${LXD_CONF2}" gen_cert_and_key "client"
+
+  # Cannot use the token with the certificates API and the correct error is returned.
+  [ "$(curl --insecure --cert "${LXD_CONF2}/client.crt" --key "${LXD_CONF2}/client.key" -X POST "https://${LXD_ADDR}/1.0/certificates" --data "{\"trust_token\": \"${tls_identity_token}\"}" | jq -er '.error')" = "Failed during search for certificate add token operation: TLS Identity token detected (you must update your client)" ]
```

Comment:
Could we use our `my_curl` wrapper like this? Despite using `--silent`, the error still shows up.

```suggestion
  [ "$(LXD_CONF="${LXD_CONF2}" my_curl -X POST "https://${LXD_ADDR}/1.0/certificates" --data "{\"trust_token\": \"${tls_identity_token}\"}" | jq -er '.error')" = "Failed during search for certificate add token operation: TLS Identity token detected (you must update your client)" ]
```

Reply:
I didn't know about `my_curl`. Sorted.